This patch fixes a bug into ostream::operator<<(double) due to the wrong size
passed into the __convert_from_v method. The wrong size is then passed to
std::snprintf function, that, on uClibc, doens't handle sized 0 buffer.

Signed-off-by: Carmelo Amoroso <carmelo.amoroso@st.com>

Index: gcc-4.2.3/libstdc++-v3/include/bits/locale_facets.tcc
===================================================================
--- gcc-4.2.3.orig/libstdc++-v3/include/bits/locale_facets.tcc	2008-01-05 12:04:43.000000000 +0100
+++ gcc-4.2.3/libstdc++-v3/include/bits/locale_facets.tcc	2008-05-21 13:45:51.877288338 +0200
@@ -1145,7 +1145,7 @@
 	const int __cs_size = __fixed ? __max_exp + __prec + 4
 	                              : __max_digits * 2 + __prec;
 	char* __cs = static_cast<char*>(__builtin_alloca(__cs_size));
-	__len = std::__convert_from_v(_S_get_c_locale(), __cs, 0, __fbuf, 
+	__len = std::__convert_from_v(_S_get_c_locale(), __cs, __cs_size, __fbuf, 
 				      __prec, __v);
 #endif
 
@@ -1779,7 +1779,7 @@
       // max_exponent10 + 1 for the integer part, + 2 for sign and '\0'.
       const int __cs_size = numeric_limits<long double>::max_exponent10 + 3;
       char* __cs = static_cast<char*>(__builtin_alloca(__cs_size));
-      int __len = std::__convert_from_v(_S_get_c_locale(), __cs, 0, "%.*Lf", 
+      int __len = std::__convert_from_v(_S_get_c_locale(), __cs, __cs_size, "%.*Lf", 
 					0, __units);
 #endif
       string_type __digits(__len, char_type());
